### PR TITLE
Added a user with invalidation permissions for heroku aws resource policy

### PR DESCRIPTION
**Pull request by sedge**

### CODE REVIEW

File: AWS/cloudformation-templates/heroku-app-and-cloudfront.json [hunk 1]
```diff
@@ -145,7 +145,37 @@
...
       }
+    },
+    "DeployUser": {
+      "Type": "AWS::IAM::User",
+      "Condition": "CreateDeployAccount",
+      "Properties": {
+        "Path": "/deploy/",
+        "Policies": [{
+          "PolicyName": {
+            "Fn::Join": ["", ["S3-RW-", {"Ref":"S3Bucket"}]]
```
Comment (jbuck) [inline AWS/cloudformation-templates/heroku-app-and-cloudfront.json]: Could you change the name of this policy to something more descriptive of what it's doing?

File: AWS/cloudformation-templates/heroku-app-and-cloudfront.json [hunk 2]
```diff
@@ -145,7 +145,37 @@
           }
         }
       }
+    },
+    "DeployUser": {
+      "Type": "AWS::IAM::User",
+      "Condition": "CreateDeployAccount",
```
Comment (jbuck) [inline AWS/cloudformation-templates/heroku-app-and-cloudfront.json]: You'll need to copy-paste the condition statement in this template as well as the parameter controlling that condition statement

File: AWS/cloudformation-templates/heroku-app-and-cloudfront.json [hunk 3]
```diff
@@ -145,7 +145,37 @@
...
+        "Policies": [{
+          "PolicyName": {
+            "Fn::Join": ["", ["S3-RW-", {"Ref":"S3Bucket"}]]
+          },
+          "PolicyDocument": {
+            "Version": "2012-10-17",
+            "Statement": [{
+              "Action": ["cloudfront:CreateInvalidation"],
+              "Effect": "Allow",
+              "Resource": { "Fn::Join": ["", ["arn:aws:s3:::", { "Ref": "S3Bucket" }]]}
```
Comment (jbuck) [inline AWS/cloudformation-templates/heroku-app-and-cloudfront.json]: This needs the same `"Resource": "*"` as the other template

File: AWS/cloudformation-templates/heroku-app-and-cloudfront.json (hunk 1 above)
Comment (sedge) [inline AWS/cloudformation-templates/heroku-app-and-cloudfront.json]: Hmmm. It looks like I need to dig into IAM a little more

File: AWS/cloudformation-templates/heroku-app-and-cloudfront.json [hunk 4]
```diff
@@ -19,6 +19,12 @@
     "LogBucketName": {
       "Type": "String",
       "Description": "S3 bucket to deliver access logs to"
+    },
+    "DeployAccount": {
+      "Type": "String",
+      "AllowedValues": ["true", "false"],
+      "Default": "false",
+      "Description": "IAM user with permissions to upload files to S3 bucket. Useful if you're using CI to build websites"
```
Comment (jbuck) [inline AWS/cloudformation-templates/heroku-app-and-cloudfront.json]: Lets change the description here, since it's only used for invalidations

**Comment by jbuck** [comment]
@sedge r+, just a nit on the description of the variable

**Pull request merged**